libbeat/common/match/matcher_test.go

```diff
@@ -209,6 +210,12 @@ func TestMatchers(t *testing.T) {
 				"fail",
 			},
 		},
+		{
+			`(?i:case)`,
+			typeOf((*regexp.Regexp)(nil)),
+			[]string{"case", "Case", "CaSe", "cAsE"},
+			nil,
+		},
```

Comment:
does it make sense to include case like this?
```
`(c[aA]se)`,
typeOf((*regexp.Regexp)(nil)),
[]string{"case", "cAse"},
[]string{"Case", "CaSe", "cAsE"},
```

to verify matcher works without `?i` after changes in future?

Reply:
Added some more tests